### PR TITLE
feat(query): add 'Global Schema Definition Not Being Used' query to Swagger

### PR DESCRIPTION
**Pull request by joaoReigota1**
Signed-off-by: João Reigota <joao.reigota@checkmarx.com>

Closes #

**Proposed Changes**
- Added Global Schema Definition Not Being Used query to Swagger

All global schemas definitions should be in use

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/libraries/openapi/library.rego [hunk 1]
```diff
@@ -248,3 +248,9 @@ is_mimetype_valid(content) {
 	known_prefixs := {"application", "audio", "font", "example", "image", "message", "model", "multipart", "text", "video"}
 	count({x | prefix := known_prefixs[x]; startswith(content, prefix)}) > 0
 }
+
+check_defenitions(doc, object, name) {
```
Comment (rogeriopeixotocx) [inline assets/libraries/openapi/library.rego:252]:
```suggestion
check_definitions(doc, object, name) {
```
small typo

File: assets/queries/openAPI/2.0/unused_schema_definition/query.rego [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) == "2.0"
+
+	doc.definitions[name]
+	openapi_lib.check_defenitions(doc, "definitions", name)
```
Comment (rogeriopeixotocx) [inline assets/queries/openAPI/2.0/unused_schema_definition/query.rego:10]:
```suggestion
	openapi_lib.check_definitions(doc, "definitions", name)
```

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]

File: assets/queries/openAPI/2.0/unused_schema_definition/test/negative1.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        "parameters": [
+          {
+            "name": "category",
+            "in": "body",
+            "description": "max records to return",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Category"
+            },
+            "desc": {
```
Comment (rafaela-soares) [inline assets/queries/openAPI/2.0/unused_schema_definition/test/negative1.json:26]: `desc` is not a valid value. Maybe we could delete it. The same happens in the other samples

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**